improve documentation for splitStrong, fanOut, splitChoice and fanIn

marcosh opened the pull request:
mention explicitly that there could be profunctors with instances for Category and Strong but not for Arrow. And similarly for ArrowChoice.

Provide also like to counter-example in https://www.eyrie.org/~zednenem/2017/07/twist

RyanGlScott commented on src/Data/Profunctor/Strong.hs:
```diff
@@ -111,7 +111,11 @@ lensVL l = dimap (getCompose #. l (\a -> Compose (a, id))) (uncurry (flip id)) .
--- `splitStrong` is analogous to `Control.Arrow.***`.
+-- `splitStrong` has a type signature similar to `Control.Arrow.***`, but using
+-- it to define an `Arrow` instance could lead to an unlawful instance.
+-- A counter-example of a profunctor `p` having instances for `Category` and
+-- `Strong` but with an unlawful instance for `Arrow` is presented in
+-- https://www.eyrie.org/~zednenem/2017/07/twist
```
```suggestion
-- A counter-example of a profunctor `p` having instances for `Category` and
-- `Strong` but with an unlawful instance for `Arrow` is presented in
-- https://www.eyrie.org/~zednenem/2017/07/twist.
```

RyanGlScott commented on src/Data/Profunctor/Strong.hs:
```diff
@@ -120,7 +124,10 @@ splitStrong l r = first' l . second' r
--- `fanOut` is analogous to `Control.Arrow.&&&`
+-- `fanOut` has a type signature similar to `Control.Arrow.&&&`, but bear in
+-- mind that there could be profunctors `p` having instances for `Category` and
+-- `Strong` but with an unlawful instance for `Arrow`. A counter-example is
+-- presented in https://www.eyrie.org/~zednenem/2017/07/twist
```
```suggestion
-- `Strong` but with an unlawful instance for `Arrow`. A counter-example is
-- presented in https://www.eyrie.org/~zednenem/2017/07/twist.
```

RyanGlScott reviewed (commented):
Thanks for the patch! The CI is failing for reasons unrelated to this PR. I've fixed the CI issues on `main`, so rebasing should resolve that.

marcosh commented:
> Thanks for the patch! The CI is failing for reasons unrelated to this PR. I've fixed the CI issues on `main`, so rebasing should resolve that.

rebased now, thanks for the review!
